plugins/announcements-backend/src/plugin.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  createBackendPlugin,
+  coreServices,
+} from '@backstage/backend-plugin-api';
+import { createRouter } from './service/router';
+import { initializePersistenceContext } from './service/persistence/persistenceContext';
+import { loggerToWinstonLogger } from '@backstage/backend-common';
+
+export const announcementsPlugin = createBackendPlugin({
+  pluginId: 'announcements',
+  register(env) {
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        http: coreServices.httpRouter,
+        permissions: coreServices.permissions,
+        database: coreServices.database,
+      },
+      async init({ http, logger, permissions, database }) {
+        http.use(
+          await createRouter({
+            permissions: permissions,
+            logger: loggerToWinstonLogger(logger),
```

Comment:
Nice, good catch. 

I missed this in my recommended code earlier and needed to be reminded this utility function existed. 🎉

Reply:
We are quite good with this part of the migration. Regarding [AnnouncementCollatorFactory.ts](https://github.com/procore-oss/backstage-plugin-announcements/blob/main/plugins/announcements-backend/src/search/AnnouncementCollatorFactory.ts), I'm still wondering if we need to adapt or migrate this part as well

Reply:
I asked Backstage maintainers about this topic. The answer confirmed my thought that we have to adapt this part:

> Hello! Yes the search integration would need to be exposed as a [module](https://backstage.io/docs/backend-system/building-plugins-and-modules/index#modules), with a bit more information on that in the [search docs](https://backstage.io/docs/features/search/how-to-guides#how-to-migrate-your-backend-installation-to-use-search-together-with-the-new-backend-system). Here's an example for a search module that adds a collator for the explore plugin: https://github.com/backstage/backstage/blob/master/plugins/search-backend-module-explore/src/alpha.ts#L37

Reply:
@gaelgoth so the recommendation is a new plugin for the collators? Something like `search-backend-module-announcements`?

If this is correct, can we do it in a separate PR or is it required for the migration?

Reply:
Yes, we can do an other branch for search module, it's not needed for the migration

Reply:
Let's merge this and add a follow-up issue to track the migration of the search module. I can pick up that ticket if needed.

Reply:
Hey @kurtaking, 

Does this PR need a changeset? Or do you want to do it when the collector module is ready

> Let's merge this and add a follow-up issue to track the migration of the search module. I can pick up that issue if needed.

Yes, please you can take this point please. I haven't had much availability lately. But I'm keeping an eye on this issue

Reply:
Yes, lets add a changset so end users know we added support for the new backend system